### PR TITLE
🔧 [MAJOR] Use Prettier

### PR DESCRIPTION
**Pull request by markbrouch**
This PR updates config rules to be [Prettier](https://github.com/prettier/prettier) compatible.

https://github.com/iambrandonn/eslint-plugin-prettier-rules/blob/master/lib/index.js

### CODE REVIEW

**Comment by markbrouch** [comment]
cc/ @ryan-roemer

File: rules/eslint/best-practices/on.js [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
     // enforce return statements in callbacks of array’s methods
     "array-callback-return": 0,
     // treat var statements as if they were block scoped
     "block-scoped-var": 0,
     // specify the maximum cyclomatic complexity allowed in a program
-    "complexity": [2, 11],
+    complexity: [2, 11],
     // require return statements to either always or never specify values
     "consistent-return": 2,
     // specify curly brace conventions for all control statements
```
Comment (Nehle) [inline rules/eslint/best-practices/on.js]: Update this doc line to reflect the new rule

File: rules/eslint/best-practices/on.js [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
-    "curly": [2, "all"],
+    curly: [2, "multi-line"],
     // require default case in switch statements
     "default-case": 0,
     // enforces consistent newlines before or after dots
```
Comment (Nehle) [inline rules/eslint/best-practices/on.js]: Update this doc line to reflect the new rule

File: rules/eslint/best-practices/on.js (hunk 1 above)
Comment (markbrouch) [inline rules/eslint/best-practices/on.js]: @Nehle Not sure what you mean - the new rule specifies that object properties should not have quotes unless needed (such as if they contain a `-`).

File: rules/eslint/best-practices/on.js (hunk 2 above)
Comment (markbrouch) [inline rules/eslint/best-practices/on.js]: @Nehle same here

**Comment by markbrouch** [comment]
@Nehle Updated code documentation you pointed out, plus a few others that needed it in https://github.com/walmartlabs/eslint-config-walmart/pull/25/commits/96c9a431d337514bde2dd3987bdcf69c4203b7e2

File: rules/eslint/errors/on.js [hunk 3]
```diff
@@ -28,8 +28,8 @@ module.exports = {
     "no-ex-assign": 2,
     // disallow double-negation boolean casts in a boolean context
     "no-extra-boolean-cast": 2,
-    // disallow unnecessary parentheses
-    "no-extra-parens": 2,
+    // allow unnecessary parentheses
```
Comment (ryan-roemer) [inline rules/eslint/errors/on.js]: NIT: I wouldn't change the comment. The rule is to "disallow". You're just turning it off with `0`.

File: rules/eslint/es6/on.js [hunk 4]
```diff
@@ -1,11 +1,11 @@
 "use strict";
 
 module.exports = {
-  "rules": {
+  rules: {
     // require braces in arrow function body
     "arrow-body-style": 0,
-    // require parens in arrow function arguments
-    "arrow-parens": 2,
+    // require parens in arrow functions with more than one param
```
Comment (ryan-roemer) [inline rules/eslint/es6/on.js]: NIT: Same comment -- don't change the comments for value changes.

File: rules/eslint/style/on.js [hunk 5]
```diff
@@ -114,30 +114,30 @@ module.exports = {
     "operator-linebreak": 0,
     // enforce padding within blocks
     "padded-blocks": 0,
-    // require quotes around object literal property names
-    "quote-props": 0,
+    // require quotes around object literal property names only when needed
```
Comment (ryan-roemer) [inline rules/eslint/style/on.js]: Same comment.

**Review by ryan-roemer — APPROVED** [review]
Other than a few nits, changes LGTM.

Love to hear how the prettier experiment goes!!!

**Pull request merged**